controllers/utils/volumegroupcontent.go

```diff
@@ -233,24 +253,65 @@ func vgcRetryOnConflictFunc(client client.Client, vgc *volumegroupv1.VolumeGroup
 	return err
 }
 
-func UpdateStaticVGC(client client.Client, vg *volumegroupv1.VolumeGroup,
+func UpdateStaticVGCFromVG(client client.Client, vg *volumegroupv1.VolumeGroup, vgClass *volumegroupv1.VolumeGroupClass, logger logr.Logger) error {
+	vgc, err := GetVGC(client, logger, *vg.Spec.Source.VolumeGroupContentName, vg.Namespace)
+	if err != nil {
+		return err
+	}
+	vgc.Spec.VolumeGroupRef = generateObjectReference(vg)
+	updateStaticVGCSpec(vgClass, vgc)
+	if err = UpdateObject(client, vgc); err != nil {
+		return err
+	}
+	return nil
+}
+
+func UpdateStaticVGC(client client.Client, vgcNamespace, vgcName string,
 	vgClass *volumegroupv1.VolumeGroupClass, logger logr.Logger) error {
-	vgc, err := GetVGC(client, logger, GetStringField(vg.Spec.Source, "VolumeGroupContentName"), vg.Name, vg.Namespace)
+	vgc, err := GetVGC(client, logger, vgcName, vgcNamespace)
 	if err != nil {
 		return err
 	}
-	updateStaticVGCSpec(vgClass, vgc, vg)
+	updateStaticVGCSpec(vgClass, vgc)
 	if err = UpdateObject(client, vgc); err != nil {
 		return err
 	}
 	return nil
 }
 
-func updateStaticVGCSpec(vgClass *volumegroupv1.VolumeGroupClass, vgc *volumegroupv1.VolumeGroupContent, vg *volumegroupv1.VolumeGroup) {
+func updateStaticVGCSpec(vgClass *volumegroupv1.VolumeGroupClass, vgc *volumegroupv1.VolumeGroupContent) {
 	secretName, secretNamespace := GetSecretCred(vgClass)
-	vgClassName := GetStringField(vg.Spec, "VolumeGroupClassName")
+	vgClassName := vgClass.Name
 	vgc.Spec.VolumeGroupClassName = &vgClassName
-	vgc.Spec.VolumeGroupRef = generateObjectReference(vg)
-	vgc.Spec.Source.Driver = vgClass.Driver
 	vgc.Spec.VolumeGroupSecretRef = generateSecretReference(secretName, secretNamespace)
+	vgc.Spec.VolumeGroupDeletionPolicy = getVolumeGroupDeletionPolicy(vgClass)
+}
+
+func UpdateThinVGC(client client.Client, vgcNamespace, vgcName string, logger logr.Logger) error {
+	vgc, err := GetVGC(client, logger, vgcName, vgcNamespace)
+	if err != nil {
+		return err
+	}
+	updateThinVGCSpec(vgc)
+	if err = UpdateObject(client, vgc); err != nil {
```

Comment:
Return err - no need to check for nil

Reply:
I prefer to leave like this.
the error handle here is a break. if we remove it and add stage after UpdateObject we could forget to break in case of an error from UpdateObject.